docs/insitu_analysis.rst

```diff
@@ -0,0 +1,140 @@
+This diagnostic outputs 2D axis-aligned projections as AMReX plotfiles prefixed with `proj`.
+
+Currently, using this diagnostic requires implementing a custom function in the problem generator for your simulation.
+(In the future, this diagnostic may be improved so that it can be configured entirely with runtime parameters.)
+
+The problem generator must call `computePlaneProjection(F const &user_f, const int dir)`
+where `user_f` is a lambda function that returns the value to project and `dir` is the axis along which the projection is taken.
+
+*Example problem generator implementation:* ::
+
+  template <> auto RadhydroSimulation<ShockCloud>::ComputeProjections(const int dir) const -> std::unordered_map<std::string, amrex::BaseFab<amrex::Real>>
+  {
+    // compute density projection
+    std::unordered_map<std::string, amrex::BaseFab<amrex::Real>> proj;
+    proj["nH"] = computePlaneProjection<amrex::ReduceOpSum>(
+        [=] AMREX_GPU_DEVICE(int i, int j, int k, amrex::Array4<const Real> const &state) noexcept {
+          Real const rho = state(i, j, k, HydroSystem<ShockCloud>::density_index);
+          return (quokka::cooling::cloudy_H_mass_fraction * rho) / m_H;
+        },
+        dir);
+    return proj;
+  }
+
+*Example input file configuration:* ::
+
+  projection_interval = 200
+  projection.dirs = x z
+
+2D Slices
+^^^^^^^^^^^^^^^^^^^^^^^
+.. note::  This is based on the *DiagFramePlane* diagnostic from PelePhysics, and the same
+  runtime parameters should apply here without modification. The output format is also the
+  same as that produced by the *Pele* codes.
+
+This outputs 2D slices of the simulation as AMReX plotfiles that can be further examined using, e.g., VisIt or yt.
+
+*Example input file configuration:* ::
+
+  quokka.diagnostics = slice_z           # Space-separated name(s) of diagnostics (arbitrary)
+  quokka.slice_z.type = DiagFramePlane   # Diagnostic type (others may be added in the future)
+  quokka.slice_z.file = slicez_plt       # Output file prefix (should end in "plt")
+  quokka.slice_z.normal = 2              # Plane normal (0 == x, 1 == y, 2 == z)
+  quokka.slice_z.center = 2.4688e20      # Coordinate in the normal direction
+  quokka.slice_z.int    = 10             # Output interval (in number of coarse steps)
+  quokka.slice_z.interpolation = Linear  # Interpolation type: Linear or Quadratic (default: Linear)
+
+  # The problem must output these derived variable(s)
+  derived_vars = temperature
+  # List of variables to include in output
+  quokka.slice_z.field_names = gasDensity gasInternalEnergy temperature
+
+Histograms/PDFs
+^^^^^^^^^^^^^^^^^^^^^^^
+.. note:: This is based on the *DiagPDF* diagnostic from PelePhysics, but significant changes
+  have been made to both the runtime parameters and the output format in order to support
+  N-dimensional histograms, log-spaced binning, and mass weighting.
+
+This adds histogram / probability density function (PDF) outputs (as fixed-width text files)
```

Comment:
You should be specific on what is computed for each of the options. That is, this should say specifically that the quantity returned is the total number of cells, total volume, or total mass in each bin, excluding cells that are masked by higher-level data.